src/control/tools/SplineHandler.h

```diff
@@ -31,7 +33,7 @@
  * The last knot and tangent can be modified using the keyboard.
  */
 
-class SplineHandler: public InputHandler {
+class SplineHandler: public InputHandler, public SnapToGridInputHandler {
```

Comment:
Don't use inheritance here, there are several reasons why not to use this:
1. The `SnapToGridInputHandler` is an object without an virtual destructor. This can lead to memory leaks.
2. Use inheritance for a "is an" relationship, this is rather a "has an" relationship.